autowarefoundation/autoware
# Feature/decision: Enhancement decision maker node

## Status
**DEVELOPMENT**

## Description

This PR is enhancement DecisionMaker.
Previous "DecisionMaker" did not make decisions, only recognition state and state management.
autowarefoundation/autoware_ai#1049

With this PR, "DecisionMaker" will be able to control the vehicle according to states such as stop, pause, left lamp on and so on.
In addition, state recognition side such as detection of stop line and correction of judgment of right/left turn is also improvement.

This DecisionMaker module makes some changes to other packages such as autoware_msg, lane_select, waypoint_loader.
But, there are made to enable/disable by changing parameter in runtime manager.
Currently, default param is disable.

I shown Tutorial video below:
[![DecisionMaker tutorial](https://img.youtube.com/vi/77J50DJoo3M/0.jpg)](https://www.youtube.com/watch?v=77J50DJoo3M)

## Requirement
- vectormap is necessary.

## Test way:
1. Load pointcloud maps, vector maps, tf(world<-map)
1. Load waypoint with check off "disableDecisionMaker" param on waypoint_loader.
1. Launch DecisionMaker
1. The rests are the same as usual self-driving.
op. You can check forced change of state by pressing button such as stop or pause on the State tab.

## Result:
The car(waypoints) is controlled according to statemachine.

## Tests
- [x] Tests with ubuntu 16.04 by yk-fujii
- [x] Tests with ubuntu 14.04 on docker by yk-fujii

- [ ] Check by others

## Todo
